public/css/theme.css

```diff
@@ -626,12 +620,16 @@ pre > code.sourceCode > span::after {
   position: absolute;
   background-color: var(--color-code-highlight-bg);
   z-index: -1;
-  left: calc(var(--line-numbers-width) - var(--code-block-side-padding));
-  right: calc(var(--line-numbers-negative-width) - var(--code-block-side-padding));
+  left: calc(-1 * var(--code-block-side-padding));
+  right: calc(-1 * var(--code-block-side-padding));
   top: 0;
   height: 100%;
 }
 
+pre.numberSource > code.sourceCode > span::after {
+  left: calc(var(--line-numbers-width) - var(--code-block-side-padding));
+}
+
```

Comment:
Only use `--line-numbers-width` in the left/right of the line highlight if we're in a `.numberSource` block, i.e. that the line numbers are actually there taking up space.